packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/data_type/mod.rs

```diff
@@ -79,36 +82,55 @@ impl<C: AsClient> DataTypeStore for PostgresStore<C> {
         Ok(identifier)
     }
 
-    async fn get_data_type(
-        &self,
-        query: &DataTypeQuery,
-    ) -> Result<Vec<DataTypeRootedSubgraph>, QueryError> {
+    async fn get_data_type(&self, query: &DataTypeQuery) -> Result<Subgraph, QueryError> {
         let DataTypeQuery {
             ref expression,
-            data_type_query_depth,
+            query_resolve_depths,
         } = *query;
 
-        stream::iter(Read::<PersistedDataType>::read(self, expression).await?)
-            .then(|data_type| async move {
-                let mut referenced_data_types = DependencyMap::new();
-
-                self.get_data_type_as_dependency(
-                    data_type.identifier.uri(),
-                    DataTypeDependencyContext {
-                        referenced_data_types: &mut referenced_data_types,
-                        data_type_query_depth,
-                    },
-                )
+        let roots_and_vertices =
+            stream::iter(Read::<PersistedDataType>::read(self, expression).await?)
+                .then(|data_type| async move {
+                    let mut referenced_data_types = DependencyMap::new();
+
+                    self.get_data_type_as_dependency(
+                        data_type.identifier.uri(),
+                        DataTypeDependencyContext {
+                            referenced_data_types: &mut referenced_data_types,
+                            data_type_query_depth: query_resolve_depths.data_type_resolve_depth,
+                        },
+                    )
+                    .await?;
+
+                    let data_type = referenced_data_types
+                        .remove(data_type.identifier.uri())
+                        .expect("root was not added to the subgraph");
+
+                    return Ok::<_, Report<QueryError>>((
+                        GraphElementIdentifier::OntologyElementId(
+                            data_type.identifier.uri().clone(),
+                        ),
+                        (
+                            GraphElementIdentifier::OntologyElementId(
+                                data_type.identifier.uri().clone(),
+                            ),
+                            Vertex::DataType(data_type),
+                        ),
+                    ));
+                })
+                .try_collect::<Vec<_>>()
                 .await?;
 
-                let root = referenced_data_types
-                    .remove(data_type.identifier.uri())
-                    .expect("root was not added to the subgraph");
+        let (roots, vertices) = roots_and_vertices.into_iter().unzip();
```

Comment:
It's a little bit annoying, that we need to collect into a vector first, and then just unzip it. I haven't found a good other solution, though, as a function like `try_unzip` does not exist.